### ISSUE TITLE
"Zoom to saved feature" zooms incorrectly on OSM maps

### ISSUE THREAD

**Issue by zestyping** [getodk/collect#2304]
#### Software and hardware versions 
ODK Collect v1.15.0

#### Problem description
The "Zoom to saved feature" function zooms to a view with the feature markers off to one side of the map, instead of zooming in on the feature (the entered polygon or trace) as it should.

#### Steps to reproduce the problem
- Make a form with a GeoShape or GeoTrace question.
- Ensure that the OSM Mapping SDK is selected in the Settings.
- Open the question.
- Long-press on the map a few times to add some markers and draw a trace or polygon.
- Tap the crosshair button.
- Choose "Zoom to saved feature".

#### Expected behavior
- The view should zoom to show the whole polygon.  Instead, it zooms out and puts the polygon off to one side.

#### Other information 
This problem does not occur in the activities that use the Google Maps SDK.

**Comment by zestyping** [comment]
This is a view of the map just after pressing "Zoom to saved feature".  The markers are over on the left edge together with the person icon.

![screenshot_20180615-104344](https://user-images.githubusercontent.com/236086/41484141-827d76b2-7090-11e8-9e59-c1bebc9f1563.png)

**Comment by lognaturel** [comment]
Well that's not terribly useful.

This seems like a regression but I'm not sure what change would have led to it. Maybe an osmdroid update? @mmarciniak90 might be good to add to your test cases if you don't have it already.

**Comment by mmarciniak90** [comment]
@zestyping could you add information about the Android version? Honestly, I'm not able to reproduce this problem.
@lognaturel added!

**Comment by zestyping** [comment]
This is on Android 6.0.1.

**Comment by zestyping** [comment]
I tried this again with the latest version from Google Play, ODK Collect 1.15.1, and the problem still occurs.

The exact data points are visible on the widget screens:

![screenshot_20180621-233622](https://user-images.githubusercontent.com/236086/41761471-3ede2b56-75ac-11e8-8fac-37244ca8b45b.png)

![screenshot_20180621-233710](https://user-images.githubusercontent.com/236086/41761476-43b18e0c-75ac-11e8-9ca5-f83af20a59f1.png)

**Comment by lognaturel** [comment]
Can't reproduce on Galaxy Tab A, Android 5.1.1 or Moto G4 running Android 7.0. I don't have an Android 6.0.1 device.

**Comment by zestyping** [comment]
Reproduced the problem again on an Android 5.1 phone, with a fresh install of ODK Collect 1.15.1 from Google Play.

I wonder if I've described the bug poorly?  I'm tapping the grey circle-and-crosshair button on the right (it's the first of four buttons in the GeoShape activity, or the second of five buttons in the GeoTrace activity).  This brings up a dialog with two buttons, "Zoom to current location" and "Zoom to saved feature", and I tap "Zoom to saved feature".
